File: app/src/main/java/ilgulee/com/modernizepractice/AutomobileRecyclerAdapter.java
package ilgulee.com.modernizepractice;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

public class AutomobileRecyclerAdapter extends RecyclerView.Adapter<AutomobileRecyclerAdapter.AutomobileViewHolder> {
    private List<Automobile> automobiles = new ArrayList<>();

    public void addAutomobile(Automobile automobile) {
        this.automobiles.add(automobile);
        notifyDataSetChanged();
    }

    public void updateData(List<Automobile> automobiles) {
        this.automobiles.addAll(automobiles);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public AutomobileViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        Context context = viewGroup.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.list_item, viewGroup, false);
        return new AutomobileViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull AutomobileViewHolder automobileViewHolder, int i) {
        Automobile automobile = automobiles.get(i);
        automobileViewHolder.textMaker.setText(automobile.getMaker());
        automobileViewHolder.textModel.setText(automobile.getModel());
    }

    @Override
    public int getItemCount() {
        return automobiles.size();
    }

    public class AutomobileViewHolder extends RecyclerView.ViewHolder {
        public TextView textMaker;
        public TextView textModel;

        public AutomobileViewHolder(@NonNull View itemView) {
            super(itemView);
            textMaker = itemView.findViewById(R.id.maker_text);
            textModel = itemView.findViewById(R.id.model_text);
        }
    }
}

File: app/src/main/java/ilgulee/com/modernizepractice/AutomobileRecyclerAdapter.kt
package ilgulee.com.modernizepractice

import android.app.Activity
import android.arch.lifecycle.Lifecycle
import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import ilgulee.com.modernizepractice.room.AutomobileDatabase

class AutomobileRecyclerAdapter(private val context: Context, val lifecycle: Lifecycle) : RecyclerView.Adapter<AutomobileRecyclerAdapter.AutomobileViewHolder>() {

    private var automobiles = ArrayList<Automobile>()

    fun setAutomobiles(arrayList: ArrayList<Automobile>) {
        this.automobiles.clear()
        this.automobiles.addAll(arrayList)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): AutomobileViewHolder {
        val context = viewGroup.context
        val inflater = LayoutInflater.from(context)
        val view = inflater.inflate(R.layout.list_item, viewGroup, false)
        return AutomobileViewHolder(view)
    }

    override fun onBindViewHolder(automobileViewHolder: AutomobileViewHolder, i: Int) {
        val automobile = automobiles[i]
        automobileViewHolder.textMaker.text = automobile.maker
        automobileViewHolder.textModel.text = automobile.model
        automobileViewHolder.deleteButton.setOnClickListener {
            Thread {
                AutomobileDatabase.getDatabase(context)!!.automobileDao().deleteAuto(automobile)
                (context as Activity).runOnUiThread {
                    automobiles.remove(automobile)
                    notifyDataSetChanged()
                }
            }.start()
        }
    }

    override fun getItemCount() = automobiles.size

    inner class AutomobileViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var textMaker: TextView
        var textModel: TextView
        var updateButton: Button
        var deleteButton: Button

        init {
            textMaker = itemView.findViewById(R.id.maker_text)
            textModel = itemView.findViewById(R.id.model_text)
            updateButton = itemView.findViewById(R.id.update_button)
            deleteButton = itemView.findViewById(R.id.delete_button)
        }

    }
}
